server.js

```diff
@@ -1,23 +1,107 @@
 
-logger.setLevel(config.logger.level);
+
+const userConfigFile = process.env.K2V_APP_USER_CONFIG_FILE || 'production.conf';
+loadSettings('CRON', userConfigFile);
+const config = getConfig('CRON');
+
+// Logger configuration
+Logger.setVerbose(config.logger.verbose);
+Logger.setTransport('console', { level: config.logger['console.level'] });
+if (config.logger['file.enable']) {
+  Logger.setTransport('file', {
+    level: config.logger['file.level'],
+    filename: config.logger['file.filename'],
+  });
+}
+const logger = new Logger('server');
+
+const serviceStateManager = new ServiceStateManager({
+  lightship: transformObjectKeys(config.lightship, camelCase),
+});
 
 process.on('unhandledRejection', (reason) => {
-    logger.error(`Unhandled Rejection at: ${reason.stack || reason}. Bailing out!!`);
-    process.kill(process.pid, "SIGTERM");
+  logger.error(
+    `Unhandled Rejection at: ${reason.stack || reason}. Bailing out!!`
+  );
+  Utils.killApplication();
 });
 
-var cronManager = new cron.CronManager();
-cronManager.init().then(() => {
-    healthcheck.init(cronManager);
-    api.init(cronManager, healthcheck.get());
-}).catch(error => {
-    logger.error(`Cron service initialization failed (${error}). Bailing out!!`);
-    process.kill(process.pid, "SIGTERM");
+let cronManager = new cron.CronManager(serviceStateManager);
+let brokerManager = new broker.BrokerHandler(serviceStateManager);
+let dbManager = new db.DB(serviceStateManager);
+
+// Registering the services, shutdown handlers and health checkers
+serviceStateManager.registerService('kafka-cron');
+serviceStateManager.registerService('kafka-broker');
+serviceStateManager.registerService('db');
+serviceStateManager.registerShutdownHandler(cronManager.shutdownHandler.bind(cronManager));
+serviceStateManager.registerShutdownHandler(brokerManager.shutdownHandler.bind(brokerManager));
+serviceStateManager.registerShutdownHandler(dbManager.shutdownHandler.bind(dbManager));
+serviceStateManager.addHealthChecker(
+  'kafka-cron',
+  cronManager.healthChecker.bind(cronManager),
+  config.healthChecker['kafka.interval.ms'],
+);
+serviceStateManager.addHealthChecker(
+  'kafka-broker',
+  brokerManager.healthChecker.bind(brokerManager),
+  config.healthChecker['kafka.interval.ms'],
+);
+serviceStateManager.addHealthChecker(
+  'db',
+  dbManager.healthChecker.bind(dbManager),
+  config.healthChecker['kafka.interval.ms'],
+);
+
+const { routes } = require('./api')(cronManager, Logger, config);
+
+// create an instance of HTTP server
+const server = WebUtils.createServer({ logger });
+
+const { 
+    tokenParsingInterceptor,
+    beaconInterceptor,
+    jsonBodyParsingInterceptor 
+} = WebUtils.framework.interceptors;
+
+// creates an instance of Express.js already configured
+const framework = WebUtils.framework.createExpress({
+    logger,
+    server,
+    routes,
+    interceptors: [
+        tokenParsingInterceptor(),
+        // beaconInterceptor(),
```

Comment:
You can add `readinessInterceptor` e `beaconInterceptor`  both use serviceStateManager